src/libsyntax/parse/parser.rs

```diff
@@ -1198,24 +1198,29 @@ impl<'a> Parser<'a> {
                 };
 
                 let body = match p.token {
-                  token::Semi => {
-                    try!(p.bump());
-                    debug!("parse_trait_methods(): parsing required method");
-                    None
-                  }
-                  token::OpenDelim(token::Brace) => {
-                    debug!("parse_trait_methods(): parsing provided method");
-                    let (inner_attrs, body) =
-                        try!(p.parse_inner_attrs_and_block());
-                    attrs.extend(inner_attrs.iter().cloned());
-                    Some(body)
-                  }
+                    token::Semi => {
+                        try!(p.bump());
+                        if let token::DocComment(_) = p.token {
+                            // bump again if the token's a trailing doc comment
+                            if p.look_ahead(1, |t| *t == token::CloseDelim(token::Brace)) {
+                                try!(p.bump());
```

Comment:
This is where the whole thing's going on. We could either bump the parser, so that it neglects this "event" (which I've done now), or we could span an error here.